src/core/data/vecfield/vecfield_initializer.h

```diff
@@ -31,65 +31,80 @@ namespace core
-            auto& vz = v.getComponent(Component::Z);
-
-            initializeComponent_(vx, layout, x_);
-            initializeComponent_(vy, layout, y_);
-            initializeComponent_(vz, layout, z_);
+            initializeComponent_(v.getComponent(Component::X), layout, x_);
+            initializeComponent_(v.getComponent(Component::Y), layout, y_);
+            initializeComponent_(v.getComponent(Component::Z), layout, z_);
         }
 
     private:
         template<typename Field, typename GridLayout>
         void initializeComponent_(Field& field, GridLayout const& layout,
-                                  initializer::ScalarFunction<dimension> const& init)
+                                  initializer::InitFunction<1> const& init)
         {
-            auto psi_X = layout.ghostStartIndex(field, Direction::X);
-            auto pei_X = layout.ghostEndIndex(field, Direction::X);
-
-            for (auto ix = psi_X; ix <= pei_X; ++ix)
-            {
-                if constexpr (dimension > 1)
-                {
-                    auto psi_Y = layout.ghostStartIndex(field, Direction::Y);
-                    auto pei_Y = layout.ghostEndIndex(field, Direction::Y);
-
-                    for (auto iy = psi_Y; iy <= pei_Y; ++iy)
-                    {
-                        if constexpr (dimension > 2)
-                        {
-                            auto psi_Z = layout.ghostStartIndex(field, Direction::Z);
-                            auto pei_Z = layout.ghostEndIndex(field, Direction::Z);
-
-                            for (auto iz = psi_Z; iz <= pei_Z; ++iz)
-                            {
-                                auto pos = layout.fieldNodeCoordinates(field, layout.origin(), ix,
-                                                                       iy, iz);
-                                field(ix, iy, iz) = init(pos[0], pos[1], pos[2]);
-                            }
-                        }
-                        else // 2D
-                        {
-                            auto pos = layout.fieldNodeCoordinates(field, layout.origin(), ix, iy);
-                            field(ix, iy) = init(pos[0], pos[1]);
-                        }
-                    }
-                }
-                else // 1D
-                {
-                    auto pos  = layout.fieldNodeCoordinates(field, layout.origin(), ix);
-                    field(ix) = init(pos[0]);
-                }
-            }
+            auto const [ix0, ix1] = layout.ghostStartToEnd(field, Direction::X);
+            auto const [x]        = layout.ghostGrids(
+                field,
+                [&](auto const&... args) {
+                    return layout.fieldNodeCoordinates(field, layout.origin(), args...);
+                },
+                /*plus=*/1);
+            assert(x.size() == (ix1 - ix0 + 1));
+
+            auto gridPtr     = init(x);
+            auto& grid       = *gridPtr;
```

Comment:
I don't get why gridPtr is useful here

Reply:
it's what keeps the data alive

Reply:
perhaps "std::shared_ptr<core::Span<T>>" would be better instead of auto for "gridPtr"